docs/javascript/imports.md

```diff
@@ -1,118 +1,194 @@
 ```
 
-<div class="tip">Thanks to <a href="./reactivity">reactivity</a>, imports can live in a JavaScript block anywhere on the page. But by convention, imports are commonly placed at the top of source files for readability.</div>
+The code above imports the default export and is equivalent to:
 
-You can use an imported library anywhere on the page.
+```js run=false
+import {default as confetti} from "npm:canvas-confetti";
+```
+
+Depending on the package, you may want to import specific named exports, or to import everything as a namespace. For example:
+
+```js run=false
+import {rollup} from "npm:d3-array"; // a single named import
+import * as d3 from "npm:d3"; // a namespace import
+```
+
+Imported symbols can be referenced in any code block or inline expression — not only in the code block that declares the import.
 
 ```js echo
-Inputs.button("Throw confetti!", {reduce: () => confetti()})
+Inputs.button("Throw confetti! 🎉", {reduce: () => confetti()})
 ```
 
+
+<div class="tip">While imports can live in code blocks anywhere on the page, by convention imports are placed at the top of pages for readability.</div>
+
+Framework provides a variety of ways to import. When you reference `d3`, `Inputs`, `Plot` or some other built-in, you’re [implicitly importing](#implicit-imports) from npm. In addition, you can import modules explicitly from:
+
+- [npm](#npm-imports),
+- [`node_modules`](#node-imports),
+- [local paths](#local-imports), or
+- [remote URLs](#url-imports).
+
+With the exception of remote URL imports, imported modules are bundled with your project, improving performance and security. In some cases, such as stylesheets and WebAssembly modules, you may need to specify additional files to download via [`import.meta.resolve`](#dynamic-imports).
+
 ## npm imports
 
-An `npm:` specifier, as shown above, denotes that the imported library will be loaded from npm rather than from `node_modules`. You don’t have to install imported libraries beforehand — just import, and the cloud shall provide. (This convention is also [used by Deno](https://docs.deno.com/runtime/manual/node/npm_specifiers).)
+Framework downloads `npm:` imports from the [npm package registry](https://www.npmjs.com/) via the [jsDelivr CDN](https://www.jsdelivr.com/esm). Unlike [imports from `node_modules`](#node-imports), you don’t have to install `npm:` imports — just import, and the cloud shall provide. 😌
 
-Under the hood, `npm:` imports are powered by [jsDelivr’s esm.run CDN](https://www.jsdelivr.com/esm). The import above is thus equivalent to:
+By default, npm imports resolve to the latest version of the given package. Imported versions are resolved on build, or during preview after you clear your [npm cache](#self-hosting-of-npm-imports) and restart the server. To load an earlier or specific version of a package, add a [semver range](https://docs.npmjs.com/about-semantic-versioning). For example, to load major version 1 of canvas-confetti:
```

Comment:
```suggestion
By default, npm imports resolve to the latest version of the given package. Imported versions are resolved on build or during preview and cached in your [npm cache](#self-hosting-of-npm-imports). To load an earlier or specific version of a package, add a [semver range](https://docs.npmjs.com/about-semantic-versioning). For example, to load major version 1 of canvas-confetti:
```

“Imported versions are resolved on build, or during preview” reads as if build did not use the npm cache and always did a fresh resolution.